File: iosApp/iosApp/Components/StatBar.swift
import SwiftUI

// 能力条：圆角 track + 按数值分段着色（弱 / 中 / 强），一眼看出强弱。
struct StatBar: View {
    let label: String
    let value: Int32

    var body: some View {
        let ratio = min(max(Double(value) / 255.0, 0), 1)
        HStack(spacing: 6) {
            Text(label)
                .font(.system(size: 12))
                .frame(width: 96, alignment: .leading)
            GeometryReader { geo in
                ZStack(alignment: .leading) {
                    Capsule().fill(Color(.systemGray5))
                    Capsule().fill(statColor(value)).frame(width: geo.size.width * ratio)
                }
            }
            .frame(height: 8)
            Text("\(value)")
                .font(.system(size: 12, weight: .semibold))
                .frame(width: 36, alignment: .trailing)
        }
    }

    // 弱红、中黄、强绿
    private func statColor(_ v: Int32) -> Color {
        if v < 50 { return Color(red: 0.925, green: 0.333, blue: 0.333) }
        if v < 90 { return Color(red: 0.984, green: 0.722, blue: 0.251) }
        return Color(red: 0.357, green: 0.725, blue: 0.392)
    }
}

File: androidApp/src/main/kotlin/com/jetbrains/kmpapp/ui/components/StatBar.kt
package com.jetbrains.kmpapp.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * 能力条：圆角 track + 按数值分段着色（弱/中/强），一眼看出强弱。
 */
@Composable
fun StatBar(
    label: String,
    value: Int,
    modifier: Modifier = Modifier,
    max: Int = 255,
) {
    val ratio = (value.toFloat() / max).coerceIn(0f, 1f)
    Row(modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
        Text(label, fontSize = 12.sp, modifier = Modifier.width(96.dp))
        Box(
            Modifier
                .weight(1f)
                .height(8.dp)
                .clip(RoundedCornerShape(50))
                .background(MaterialTheme.colorScheme.surfaceVariant),
        ) {
            Box(
                Modifier
                    .fillMaxWidth(ratio)
                    .height(8.dp)
                    .clip(RoundedCornerShape(50))
                    .background(statColor(value)),
            )
        }
        Text(
            "$value",
            fontSize = 12.sp,
            fontWeight = FontWeight.SemiBold,
            textAlign = TextAlign.End,
            modifier = Modifier.width(36.dp),
        )
    }
}

// 弱红、中黄、强绿
private fun statColor(value: Int): Color = when {
    value < 50 -> Color(0xFFEC5555)
    value < 90 -> Color(0xFFFBB840)
    else -> Color(0xFF5BB964)
}
